Xamarin.PropertyEditing.Mac/Controls/BasePathEditorControl.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Collections;
+using AppKit;
+using Foundation;
+using Xamarin.PropertyEditing.Mac.Resources;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal abstract class BasePathEditorControl<T> : PropertyEditorControl<PropertyViewModel<T>>
+	{
+		protected readonly TextFieldSmallButtonContainer currentTextField;
+
+		protected readonly NSOpenPanel panel;
+		protected readonly SmallButton browsePathButton;
+		protected readonly SmallButton revealPathButton;
+
+		public override NSView FirstKeyView => this.currentTextField;
+		public override NSView LastKeyView => this.revealPathButton.Enabled ? this.revealPathButton : this.browsePathButton;
+
+		protected BasePathEditorControl (IHostResourceProvider hostResource)
+			: base (hostResource)
+		{
+			this.currentTextField = new TextFieldSmallButtonContainer ();
+			this.currentTextField.Changed += CurrentTextField_Changed;
+			AddSubview (this.currentTextField);
+
+			#region Reveal handler
+
+			this.revealPathButton = new SmallButton {
+				Font = NSFont.FromFontName (DefaultFontName, DefaultFontSize),
+				StringValue = string.Empty,
+				Image = hostResource.GetNamedImage ("path-reveal"),
+			};
+
+			this.currentTextField.AddButton (this.revealPathButton);
+
+			this.panel = new NSOpenPanel {
+				ShowsResizeIndicator = true,
+				ShowsHiddenFiles = false,
+				CanCreateDirectories = true,
+				AllowsMultipleSelection = false
+			};
+
+			// update the value on keypress
+			this.revealPathButton.Activated += OnRevealPathButtonActivated;
+
+			#endregion
+
+			#region Browse Path
+
+			this.browsePathButton = new SmallButton {
+				Font = NSFont.FromFontName (DefaultFontName, DefaultFontSize),
+				StringValue = string.Empty,
+				Image = hostResource.GetNamedImage ("path-browse"),

```

Comment:
These images need to be set in `ViewDidChangeEffectiveAppearance` so they can properly adapt for dark/light on change and the method should be called at the bottom of the `ctor`

Reply:
Fixed